src/geometry.rs

```diff
@@ -38,14 +54,26 @@ impl<T> Rect<T>
 where
     T: Add<Output = T> + Copy + Clone,
 {
+    /// The sum of the coordinates of the left and right side of the rectangle
+    ///
+    /// **NOTE:** this is *not* the width of the rectangle.
     pub(crate) fn horizontal(&self) -> T {
         self.start + self.end
     }
 
+    /// The sum of the coordinates of the bottom and top side of the rectangle
+    ///
+    /// **NOTE:** this is *not* the height of the rectangle.
     pub(crate) fn vertical(&self) -> T {
         self.top + self.bottom
     }
 
+    /// The main layout-axis sum of the rectangle
+    ///
+    /// This is always perpendicular to [`Rect::cross`].
+    ///
+    /// If the [`FlexDirection`] is [`FlexDirection::Row`] or [`FlexDirection::RowReverse`], this is [`Rect::horizontal`].
+    /// Otherwise, this is [`Rect::vertical`].
     pub(crate) fn main(&self, direction: FlexDirection) -> T {
```

Comment:
I think in the future we should rename all of these functions to make them more clear, e.g. `main_axis` or something similar